relay-server/src/services/buffer/envelope_buffer/mod.rs

```diff
@@ -162,13 +162,30 @@ where
 
     /// Returns a reference to the next-in-line envelope, if one exists.
     pub async fn peek(&mut self) -> Result<Option<&Envelope>, EnvelopeBufferError> {
+        let Some((
+            QueueItem {
+                key: stack_key,
+                value: _,
+            },
+            _,
+        )) = self.priority_queue.peek()
+        else {
+            return Ok(None);
+        };
+
+        let stack_key = *stack_key;
```

Comment:
Untested:

```suggestion
        let Some((
            QueueItem {
                key: &stack_key,
                value: _,
            },
            _,
        )) = self.priority_queue.peek()
        else {
            return Ok(None);
        };
```

Reply:
I tried already but it was not possible :(